codec-dns/src/main/java/io/netty/handler/codec/dns/DnsResponseDecoder.java

```diff
@@ -61,8 +61,15 @@ final DnsResponse decode(A sender, A recipient, ByteBuf buffer) throws Exception
             final int additionalRecordCount = buffer.readUnsignedShort();
 
             decodeQuestions(response, buffer, questionCount);
-            decodeRecords(response, DnsSection.ANSWER, buffer, answerCount);
-            decodeRecords(response, DnsSection.AUTHORITY, buffer, authorityRecordCount);
+            if (!decodeRecords(response, DnsSection.ANSWER, buffer, answerCount)) {
+                success = true;
+                return response;
+            }
+            if (!decodeRecords(response, DnsSection.AUTHORITY, buffer, authorityRecordCount)) {
+                success = true;
+                return response;
+            }
```

Comment:
Just an idea, could have `decodeRecords` throw some specific exception rather than returning false in truncation case. Then the logic here would just become:
```java
try {
    decodeRecords(...);
    decodeRecords(...);
    decodeRecords(...);
} catch (TruncationException e) {
    // return response with readable bytes remaining
}
success = true;
return response;
```

Reply:
I would prefer to keep it as it is.